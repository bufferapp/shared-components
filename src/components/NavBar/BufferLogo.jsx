import React from 'react';

const BufferLogo = () => (
  <svg width='72' height='64' viewBox='0 0 72 64' fill='none' xmlns='http://www.w3.org/2000/svg'>
    <path d='M0 0H72V64H0V0Z' fill='#2C4BFF' />
    <path d='M35.574 30.9796C35.6444 31.012 35.7286 31.0346 35.8182 31.0476C36.0255 31.0776 36.262 31.055 36.4258 30.9796L47.8241 25.7262C47.95 25.6684 48.0083 25.5908 47.9989 25.5146C47.9907 25.4488 47.9324 25.3841 47.8241 25.334L36.4258 20.0809C36.1916 19.973 35.8083 19.973 35.574 20.0809L24.1753 25.334C23.9416 25.4419 23.9416 25.6183 24.1753 25.7262L35.574 30.9796Z' fill='#fff' />
    <path d='M47.8241 31.8037L45.0938 30.5457C44.8601 30.4378 44.4768 30.4378 44.2426 30.5457L36.4258 34.1482C36.1916 34.2561 35.8083 34.2561 35.574 34.1482L27.7573 30.5457C27.523 30.4378 27.1398 30.4378 26.9055 30.5457L24.1753 31.8037C23.9416 31.9116 23.9416 32.0884 24.1753 32.1963L35.574 37.4493C35.8083 37.5572 36.1916 37.5572 36.4258 37.4493L47.8241 32.1963C48.0583 32.0884 48.0583 31.9116 47.8241 31.8037Z' fill='#fff' />
    <path d='M45.0938 37.0155L47.8241 38.2738C47.9505 38.3318 48.0088 38.41 47.9989 38.4865C47.9901 38.5521 47.9318 38.6162 47.8241 38.666L36.4258 43.9191C36.1916 44.027 35.8083 44.027 35.574 43.9191L24.1753 38.666C23.9416 38.5581 23.9416 38.3817 24.1753 38.2738L26.9055 37.0155C27.1398 36.9076 27.523 36.9076 27.7573 37.0155L35.574 40.618C35.8083 40.7259 36.1916 40.7259 36.4258 40.618L44.2426 37.0155C44.4768 36.9076 44.8601 36.9076 45.0938 37.0155Z' fill='#fff' />
  </svg>
);

export default BufferLogo;
