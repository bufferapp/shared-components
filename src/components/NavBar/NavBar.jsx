import React from 'react';

const NavBar = () => (
  <div />
);

NavBar.propTypes = {};

export default NavBar;
