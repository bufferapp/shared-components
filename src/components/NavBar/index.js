export { default } from './NavBarMenu/NavBarMenu';
