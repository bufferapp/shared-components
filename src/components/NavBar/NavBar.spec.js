import snap from 'jest-auto-snapshots';
import NavBar from './NavBar';

snap(NavBar, './NavBar.jsx');
