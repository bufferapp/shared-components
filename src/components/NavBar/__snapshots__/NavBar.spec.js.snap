// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > NavBar Matches snapshot when array prop "helpMenuItems" has one item: "[object Object]" 1`] = `
<styled.nav>
  <styled.div>
    <BufferLogo />
    <NavBarProduct
      activeProduct="publish"
    />
  </styled.div>
  <styled.div>
    <Select
      customButton={[Function]}
      hasCustomAction={false}
      hasIconOnly={false}
      hideSearch={true}
      horizontalOffset="-16px"
      isOpen={false}
      isSplit={false}
      items={
        Array [
          Object {
            "component": <NodeFixture />,
            "hasDivider": true,
            "id": "jest-auto-snapshots String Fixture",
            "onItemClick": [MockFunction],
            "title": "jest-auto-snapshots String Fixture",
          },
        ]
      }
      label=""
      searchPlaceholder="Search"
      shortcutsEnabled={true}
      size="medium"
      type="secondary"
      xPosition="right"
      yPosition="bottom"
    />
    <Select
      customButton={[Function]}
      hasCustomAction={false}
      hasIconOnly={false}
      hideSearch={true}
      horizontalOffset="16px"
      isOpen={false}
      isSplit={false}
      items={
        Array [
          Object {
            "component": <NodeFixture />,
            "hasDivider": true,
            "id": "jest-auto-snapshots String Fixture",
            "onItemClick": [MockFunction],
            "title": "jest-auto-snapshots String Fixture",
          },
        ]
      }
      label=""
      searchPlaceholder="Search"
      shortcutsEnabled={true}
      size="medium"
      type="secondary"
      xPosition="left"
      yPosition="bottom"
    />
  </styled.div>
</styled.nav>
`;

exports[`jest-auto-snapshots > NavBar Matches snapshot when array prop "helpMenuItems" is an empty array: "[]" 1`] = `
<styled.nav>
  <styled.div>
    <BufferLogo />
    <NavBarProduct
      activeProduct="publish"
    />
  </styled.div>
  <styled.div>
    <Select
      customButton={[Function]}
      hasCustomAction={false}
      hasIconOnly={false}
      hideSearch={true}
      horizontalOffset="-16px"
      isOpen={false}
      isSplit={false}
      items={Array []}
      label=""
      searchPlaceholder="Search"
      shortcutsEnabled={true}
      size="medium"
      type="secondary"
      xPosition="right"
      yPosition="bottom"
    />
    <Select
      customButton={[Function]}
      hasCustomAction={false}
      hasIconOnly={false}
      hideSearch={true}
      horizontalOffset="16px"
      isOpen={false}
      isSplit={false}
      items={
        Array [
          Object {
            "component": <NodeFixture />,
            "hasDivider": true,
            "id": "jest-auto-snapshots String Fixture",
            "onItemClick": [MockFunction],
            "title": "jest-auto-snapshots String Fixture",
          },
        ]
      }
      label=""
      searchPlaceholder="Search"
      shortcutsEnabled={true}
      size="medium"
      type="secondary"
      xPosition="left"
      yPosition="bottom"
    />
  </styled.div>
</styled.nav>
`;

exports[`jest-auto-snapshots > NavBar Matches snapshot when passed all props 1`] = `
<styled.nav>
  <styled.div>
    <BufferLogo />
    <NavBarProduct
      activeProduct="publish"
    />
  </styled.div>
  <styled.div>
    <Select
      customButton={[Function]}
      hasCustomAction={false}
      hasIconOnly={false}
      hideSearch={true}
      horizontalOffset="-16px"
      isOpen={false}
      isSplit={false}
      items={
        Array [
          Object {
            "component": <NodeFixture />,
            "hasDivider": true,
            "id": "jest-auto-snapshots String Fixture",
            "onItemClick": [MockFunction],
            "title": "jest-auto-snapshots String Fixture",
          },
        ]
      }
      label=""
      searchPlaceholder="Search"
      shortcutsEnabled={true}
      size="medium"
      type="secondary"
      xPosition="right"
      yPosition="bottom"
    />
    <Select
      customButton={[Function]}
      hasCustomAction={false}
      hasIconOnly={false}
      hideSearch={true}
      horizontalOffset="16px"
      isOpen={false}
      isSplit={false}
      items={
        Array [
          Object {
            "component": <NodeFixture />,
            "hasDivider": true,
            "id": "jest-auto-snapshots String Fixture",
            "onItemClick": [MockFunction],
            "title": "jest-auto-snapshots String Fixture",
          },
        ]
      }
      label=""
      searchPlaceholder="Search"
      shortcutsEnabled={true}
      size="medium"
      type="secondary"
      xPosition="left"
      yPosition="bottom"
    />
  </styled.div>
</styled.nav>
`;

exports[`jest-auto-snapshots > NavBar Matches snapshot when passed only required props 1`] = `
<styled.nav>
  <styled.div>
    <BufferLogo />
    <NavBarProduct />
  </styled.div>
  <styled.div>
    <Select
      customButton={[Function]}
      hasCustomAction={false}
      hasIconOnly={false}
      hideSearch={true}
      horizontalOffset="16px"
      isOpen={false}
      isSplit={false}
      items={
        Array [
          Object {
            "component": <NodeFixture />,
            "hasDivider": true,
            "id": "jest-auto-snapshots String Fixture",
            "onItemClick": [MockFunction],
            "title": "jest-auto-snapshots String Fixture",
          },
        ]
      }
      label=""
      searchPlaceholder="Search"
      shortcutsEnabled={true}
      size="medium"
      type="secondary"
      xPosition="left"
      yPosition="bottom"
    />
  </styled.div>
</styled.nav>
`;
